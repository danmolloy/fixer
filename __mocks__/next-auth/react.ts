import { mockSession } from '../session';

module.exports = {
  __esModule: true,
  ...jest.requireActual('next-auth/react'),
  auth: jest.fn(() => {
    return mockSession;
  }),
  signIn: jest.fn(),
  signOut: jest.fn(),
  useSession: jest.fn(),
};
