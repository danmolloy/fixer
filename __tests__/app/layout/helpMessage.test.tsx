import '@testing-library/jest-dom';
import HelpMessage, { HelpMessageProps } from '../../../app/layout/helpMessage';
import { screen, render } from '@testing-library/react';

const mockProps: HelpMessageProps = {
  head: 'Alert!!',
  additional: 'Beware!',
};

describe('<HelpMessage />', () => {
  beforeEach(() => {
    render(<HelpMessage {...mockProps} />);
  });
  it('<HelpMessage /> renders', () => {
    const helpMessage = screen.getByTestId('help-message');
    expect(helpMessage).toBeInTheDocument();
  });
  it('matches snapshot', () => {
    const helpMessage = screen.getByTestId('help-message');
    expect(helpMessage).toMatchSnapshot();
  });
});
