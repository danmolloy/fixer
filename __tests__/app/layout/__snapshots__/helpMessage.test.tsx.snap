// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelpMessage /> matches snapshot 1`] = `
<div
  data-testid="help-message"
>
  <p
    class="font-semibold"
  >
    Alert!!
  </p>
  <p
    class="text-sm"
  >
    Beware!
  </p>
</div>
`;
